app/controllers/DataSetController.scala

```diff
@@ -142,18 +140,40 @@ class DataSetController @Inject()(userService: UserService,
       for {
         dataSets <- dataSetDAO.findAll ?~> "dataSet.list.failed"
         filtered <- filter.applyOn(dataSets)
-        requestingUserTeamManagerMemberships <- Fox.runOptional(request.identity)(user =>
-          userService.teamManagerMembershipsFor(user._id))
-        js <- Fox.serialCombined(filtered)(
-          d =>
-            dataSetService
-              .publicWrites(d, request.identity, skipResolutions = true, requestingUserTeamManagerMemberships))
+        js <- listGrouped(filtered, request.identity)
```

Comment:
here the assumption is that there are (way) less organizations and datastores in the db than there are datasets. by the grouped handling of datasets, we only have to query organization and datastore info once (or times number-of-organizations respectively), resulting in less redundant queries

Reply:
*fewer